### DISCUSSION TITLE
Which Loss function is used for NER and Transformer tokenizer?

### DISCUSSION THREAD

**Discussion by sureshPalla230393** [explosion/spaCy#8211] [Help: Other Questions] [answered]
I'm trying to run the 'en_core_web_trf' on my own training set. The loss after each epoch doesn't seem to improve and the model is performing very bad compared to 'en_core_web_sm' . I'm not able make sense what exactly is the 'Transformer' loss shown on console along with the 'NER' loss. I would also like to know which Loss functions are used for the 'Transformer' and the 'NER'. 
![image](https://user-images.githubusercontent.com/33569360/119767220-16cfe680-bed4-11eb-882b-44c6adec5ea2.png)

**Comment by polm** [comment, marked as answer]
Sorry for the delayed reply on this. The NER loss calculation is [here](https://github.com/explosion/spaCy/blob/67d9ebc922932203b141c71d35bf3d66bbe5d3d7/spacy/scorer.py#L818), it's pretty basic but optimized for whole-entity matching rather than best-effort token overlap. 

Loss for the Transformer is just the backpropagated loss from other components in the pipeline, in your case the NER. In your case you should just be able to ignore it. You can see the source [here](https://github.com/explosion/spacy-transformers/blob/424dd3a76ffeaf0720ab09419403fa21fafd9a47/spacy_transformers/pipeline_component.py#L253).
